### ISSUE TITLE
Error filtering parameters in v0.30.0

### ISSUE THREAD

**Issue by riboher** [oblador/loki#391]
Hi there! 

I would like to report a bug after updating to `v0.30.0`. After this update, running the loki tests results in a `TypeError: Converting circular structure to JSON`  during the `FETCH_STORIES` step.

After some debugging, I've found out that the error is originated in [this chunk of code](https://github.com/oblador/loki/compare/v0.29.0...v0.30.0#diff-fc5b6032bd9075ce99acac83c9144b5f2c022b17fe2d06b0f0aaf63c5f3a2947R27). Removing that part solves the issue. I guess it has to do with some of the `parameters` I'm passing down but I've been trying to pass them in different formats and I keep seeing the same error.

I'll be happy to provide more details if needed to. Thank you!

**Comment by oblador** [comment]
Yeah, it's most likely the parameter serialization! What parameters do you pass today?

**Comment by riboher** [comment]
Probably the most problematic are those related to routing and workers. A typical component story I could have would be in the following format:

```
msw: [
    ...defaultMswHandlers,
    rest.post('/api/users', (req, res, ctx) => {
      return res(ctx.json(mockResponse));
    }),
  ],
 nextRouter: {
      asPath: '/users',
      path: '/users',
    },
```

The base parameters object is this one:

```
export const parameters: Parameters & TestAppWrapperParameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  authSession,
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  msw: defaultMswHandlers,
  nextRouter: {
    Provider: RouterContext.Provider,
    asPath: '/',
    path: '/',
    query: {},
  },
  options: {
    storySort: (a, b) =>
      a[1].kind === b[1].kind
        ? 0
        : a[1].id.localeCompare(b[1].id, undefined, { numeric: true }),
  },
};
```

However, I've tried removing the `msw` related parameters (I've read the request object is cyclic so I thought maybe that was that was causing that error), but the problem kept popping up. There's a chance I forgot removing it from some part in the app though. Either way, I hope it helps. 

Thanks for the quick response BTW! 😄

**Comment by riboher** [comment]
Wow, that was fast! Thanks so much!

**Comment by oblador** [comment]
Fixed in [0.30.1](https://github.com/oblador/loki/releases/tag/v0.30.1)

**Comment by riboher** [comment]
Yup! Just tried it. Working like a charm. Thanks again!
